src/lib/docker-utils.coffee

```diff
@@ -74,34 +92,48 @@ module.exports = class DockerUtils extends DockerToolbelt
 			request.getAsync(url, opts)
 			.get(1)
 			.then (responseBody) ->
+				token = responseBody?.token
 				opts =
 					followRedirect: false
 					timeout: requestTimeout
 
-				if responseBody?.token?
+				if token?
 					opts.auth =
-						bearer: responseBody.token
+						bearer: token
 						sendImmediately: true
 				new Promise (resolve, reject) ->
-					request.get("#{deltaEndpoint}/api/v2/delta?src=#{deltaSource}&dest=#{imgDest}", opts)
+					request.get("#{deltaEndpoint}/api/v#{version}/delta?src=#{deltaSource}&dest=#{imgDest}", opts)
 					.on 'response', (res) ->
-						res.resume() # discard response body -- we only care about response headers
 						if res.statusCode in [ 502, 504 ]
 							reject(new Error('Delta server is still processing the delta, will retry'))
-						else if not (300 <= res.statusCode < 400 and res.headers['location']?)
+						else if res.statusCode is 200 and not (300 <= res.statusCode < 400 and res.headers['location']?)
 							reject(new Error("Got #{res.statusCode} when requesting image from delta server."))
+						else if res.statusCode is 200
+							# v3
+							try
+								name = JSON.parse(res.body).name
+								resolve(applyBalenaDelta(deltaUrl, token, onProgress, log))
+							catch e
+								reject(e)
 						else
```

Comment:
Would it be better to check explicitly for the  version rather than depending on the server's response?